tools/pythonpkg/tests/fast/test_type.py

```diff
@@ -75,6 +75,7 @@ def test_struct_type(self):
         type = duckdb.struct_type({'a': BIGINT, 'b': BOOLEAN})
         assert str(type) == 'STRUCT(a BIGINT, b BOOLEAN)'
 
+        # FIXME: create an unnamed struct when fields are provided as a list
```

Comment:
Would this allow the user to create 'external' unnamed structs? I understand this PR does not allow for unnamed structs 'outside' of duckdb (only during histogram, row, etc.). So, do we still want to continue creating a named struct under the hood for a list of field types? (@Tishj )

Reply:
I'm not sure I follow, DuckDBPyType is not much more than a wrapper around a `LogicalType`, with some helper methods to construct the logical type and some methods to query data about the type (children for structs, scale/precision for decimal type etc)

Reply:
```py
>>> import duckdb 
>>> duckdb.execute('create table tbl as select ROW(1, 2, 3)')
>>> rel = duckdb.table('tbl')
>>> rel.types 
[STRUCT(v1 INTEGER, v2 INTEGER, v3 INTEGER)]
>>> duckdb.execute('insert into tbl VALUES({a: 1, b: 2, c: 3})')
>>> duckdb.table('tbl').fetchall()
[({'v1': 1, 'v2': 2, 'v3': 3},), ({'v1': 1, 'v2': 2, 'v3': 3},)]
```

Reply:
If we don't want users to have the ability to have unnamed structs in their tables then this PR would need some changes @taniabogatsch

Reply:
Yes, but is this intended behavior? If we create a table with `ROW`, do we want to continue being able to cast any struct into that struct (which has named fields after the table creation)? Or do we only have unnamed structs within one statement? That's what I am wondering about...

Reply:
Yes, this is exactly what I am wondering about.
```sql
>>> duckdb.execute('insert into tbl VALUES({a: 1, b: 2, c: 3})')
>>> duckdb.table('tbl').fetchall()
[({'v1': 1, 'v2': 2, 'v3': 3},), ({'v1': 1, 'v2': 2, 'v3': 3},)]
```
Because now it is a struct, we don't want to allow this insert. As the names don't match...? But I am not sure about this, because it would limit the usability of `ROW`...

Reply:
So currently this does work:
```sql
create table unnamed as from (VALUES (ROW(3)));

create table named (bar struct(pip int));

insert into named select * from unnamed;

insert into named VALUES (ROW(42));

select * from named;
----
{'pip': 3}
{'pip': 42}
```

But not sure if it does from python

Reply:
Yes, theres no difference there, all just sql
I think what Tania is saying that we should only allow unnamed structs as intermediate types, so creating from ROW should make the column have a named struct, with v1, v2, etc..

Reply:
@Mytherin